Services/ISelectedCurrencyProvider.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orchard;
+
+namespace Nwazet.Commerce.Services {
+    public interface ISelectedCurrencyProvider : IDependency {
+        /// <summary>
+        /// The name of the provider is used to identify uniquely the provider.
+        /// </summary>
+        string Name { get; }
```

Comment:
Once we have a currency class, we can return an instance of that instead of having separate methods and properties to get name, symbol, etc.